thoth-errors/src/database_errors.rs

```diff
@@ -0,0 +1,268 @@
+
+use crate::ThothError;
+
+/// A map of database constraint name and a corresponding error to output
+/// when the constraint is violated.
+///
+/// To obtain a list of unique and check constraints:
+/// ```sql
+/// SELECT conname
+/// FROM pg_catalog.pg_constraint con
+/// INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+/// WHERE nsp.nspname = 'public'
+/// AND contype in ('u', 'c');
+/// ```
+static DATABASE_CONSTRAINT_ERRORS: Map<&'static str, &'static str> = phf_map! {
+    "publisher_uniq_idx" => "A publisher with this name already exists.",
+    "imprint_uniq_idx" => "An imprint with this name already exists.",
+    "doi_uniq_idx" => "A work with this DOI already exists.",
+    "language_uniq_work_idx" => "Duplicate language code.",
+    "series_issn_print_idx" => "A series with this ISSN already exists.",
+    "series_issn_digital_idx" => "A series with this ISSN already exists.",
+    "issue_uniq_ord_in_series_idx" => "An issue with this ordinal number already exists.",
+    "orcid_uniq_idx" => "A contributor with this ORCID ID already exists.",
+    "location_uniq_canonical_true_idx" => "A canonical location for this publication already exists.",
+    "location_uniq_platform_idx" => "A location on the selected platform already exists.",
+    "email_uniq_idx" => "An account with this email already exists.",
+    "affiliation_uniq_ord_in_contribution_idx" => "An affiliation with this ordinal number already exists.",
+    "contribution_contribution_ordinal_work_id_uniq" => "A contribution with this ordinal number already exists.",
+    "contribution_work_id_contributor_id_contribution_type_uniq" => "A contribution of this type already exists for this contributor.",
+    "issue_series_id_work_id_uniq" => "An issue on the selected series already exists for the this work.",
+    "publication_publication_type_work_id_uniq" => "A publication with the selected type already exists.",
+    "work_relation_ordinal_type_uniq" => "A relation with this ordinal number already exists.",
+    "work_relation_relator_related_uniq" => "A relation between these two works already exists.",
+    "affiliation_affiliation_ordinal_check" => "An affiliation ordinal number must be greater than 0.",
+    "contribution_contribution_ordinal_check" => "A contribution ordinal number must be greater than 0.",
+    "affiliation_position_check" => "Position must not be an empty string.",
+    "contribution_biography_check" => "Biography must not be an empty string.",
+    "contribution_first_name_check" => "First name must not be an empty string.",
+    "contribution_full_name_check" => "Full name must not be an empty string.",
+    "contribution_last_name_check" => "Last name must not be an empty string.",
+    "contributor_first_name_check" => "First name must not be an empty string.",
+    "contributor_full_name_check" => "Full name must not be an empty string.",
+    "contributor_last_name_check" => "Last name must not be an empty string.",
```

Comment:
We should use "Given name" and "Family name" instead of "First name" and "Last name" as this is user-facing. Ensure field names in other error strings match what the user is shown.

Reply:
This error shouldn't actually be displayed in the APP as we prevent empty strings. In this case this error is most likely only going to be output to API users, who'll need to know the actual field (which we should update)